.github/workflows/ci.yml

```diff
@@ -59,7 +59,7 @@ jobs:
                       /tmp/.uv-cache
                       ./.venv
                       ./.check.md5
-                  key: env-cache:main
+                  key: env-cache:main:${{ hashFiles('.python-version') }}
```

Comment:
this change can now be reverted becaue `env-cache:main` was built using python 3.12. Apologies for this!